pkg/execution/history/lifecycle.go

```diff
@@ -526,6 +526,14 @@ func (l lifecycle) OnSleep(
 	}
 }
 
+// Do nothing
+func (lifecycle) OnConcurrencyLimit(
+	context.Context,
```

Comment:
Are these params enough for the consumer to know which level of concurrency was hit? Like whether it was account or workflow concurrency, and eventually workspace concurrency

Reply:
I thought a little about it, but metrics only tell what happens, not why.
Which concurrency being hit should have a separate indicator, because that's the why.
This is good enough, at least for now until we learn otherwise.